httpsender/add-more-headers.js

```diff
@@ -0,0 +1,64 @@
+// This HttpSender script adds headers to all messages transmitted by zaproxy,
+// including automated tools. Refer to the HttpSender class definition:
+// https://github.com/zaproxy/zaproxy/blob/master/zap/src/main/java/org/parosproxy/paros/network/HttpSender.java
+// for a list of 'initiator' values (although we don't use them).
+
+var ScriptVars = Java.type("org.zaproxy.zap.extension.script.ScriptVars");
+
+/*
+ * HttpSender scripts do not support parameters, so we'll use a known global
+ * variable to supply desired content. The value of this variable should be a
+ * JSON string containing a serialized map<String, String> object. The map keys
+ * are the desired header name and the values are the header values.
+ */
+
+PARAMETER_VARIABLE = "add_headers";
+user_headers = null;
+
+// Logging with the script name is super helpful!
+function logger() {
+	print('[' + this['zap.script.name'] + '] ' + arguments[0]);
+}
+
+// Parse and store headers where we can get at them quickly
+function initializeHeaders(variableName) {
+	logger("Initializing...");
+	user_headers = JSON.parse(ScriptVars.getGlobalVar(variableName));
```

Comment:
How is the global variable being set? Wondering if this could use custom variable (thus get an `Object` instead of a `String`).

Reply:
We're doing it like this (from python) at present:

```python
    # Store the result in a global variable where it can be accessed later
    header_json = json.dumps(header_dict)
    log.debug("Setting user headers: %s", header_json)
    zap.script.set_global_var("add_headers", header_json)
```

The `zap` object is instantiated by `zapv2.ZAPv2()`.

Reply:
OK, from the API needs to be as is then.